src/langchain_google_alloydb_pg/async_checkpoint.py

```diff
@@ -0,0 +1,595 @@
+            if writes
+            else []
+        )
+
+    def _search_where(
+        self,
+        config: Optional[RunnableConfig],
+        filter: MetadataInput,
+        before: Optional[RunnableConfig] = None,
+    ) -> tuple[str, dict[str, Any]]:
+        """Return WHERE clause predicates for alist() given config, filter, before.
+
+        This method returns a tuple of a string and a tuple of values. The string
+        is the parameterized WHERE clause predicate (including the WHERE keyword):
+        "WHERE column1 = $1 AND column2 IS $2". The list of values contains the
+        values for each of the corresponding parameters.
+        """
+        wheres = []
+        param_values = {}
+
+        # construct predicate for config filter
+        if config:
+            wheres.append("thread_id = :thread_id")
+            param_values.update({"thread_id": config["configurable"]["thread_id"]})
+            checkpoint_ns = config["configurable"].get("checkpoint_ns")
+            if checkpoint_ns is not None:
+                wheres.append("checkpoint_ns = :checkpoint_ns")
+                param_values.update({"checkpoint_ns": checkpoint_ns})
+
+            if checkpoint_id := get_checkpoint_id(config):
+                wheres.append("checkpoint_id = :checkpoint_id")
+                param_values.update({"checkpoint_id": checkpoint_id})
+
+        # construct predicate for metadata filter
+        if filter:
+            wheres.append("encode(metadata,'escape')::jsonb @> :metadata ")
+            param_values.update({"metadata": f"{json.dumps(filter)}"})
+
+        # construct predicate for `before`
+        if before is not None:
+            wheres.append("checkpoint_id < :checkpoint_id")
+            param_values.update({"checkpoint_id": get_checkpoint_id(before)})
+
+        return (
+            "WHERE " + " AND ".join(wheres) if wheres else "",
+            param_values,
+        )
+
+    async def aput(
+        self,
+        config: RunnableConfig,
+        checkpoint: Checkpoint,
+        metadata: CheckpointMetadata,
+        new_versions: ChannelVersions,
+    ) -> RunnableConfig:
+        """Asynchronously store a checkpoint with its configuration and metadata.
+
+        Args:
+            config (RunnableConfig): Configuration for the checkpoint.
+            checkpoint (Checkpoint): The checkpoint to store.
+            metadata (CheckpointMetadata): Additional metadata for the checkpoint.
+            new_versions (ChannelVersions): New channel versions as of this write.
+
+        Returns:
+            RunnableConfig: Updated configuration after storing the checkpoint.
+        """
+        configurable = config["configurable"].copy()
+        thread_id = configurable.pop("thread_id")
+        checkpoint_ns = configurable.pop("checkpoint_ns")
+        checkpoint_id = configurable.pop(
+            "checkpoint_id", configurable.pop("thread_ts", None)
+        )
+
+        next_config: RunnableConfig = {
```

Comment:
I don't follow this logic -- why is there a copy/pop? configureable doesn't seem to be used anywhere